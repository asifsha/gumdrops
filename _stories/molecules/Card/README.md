The `<Card>` component is a standard container component, designed to hold groups of related information. Cards are usually arranged in a uniform grid.
