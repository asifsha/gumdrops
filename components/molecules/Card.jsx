import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Card = ({ option, size, className, style, children }) => {
    const baseClass = 'gds-card';
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${option}`]: option,
        [`${baseClass}--${size}`]: size
    });

    return (
        <div className={rootClass} style={style}>
            {children}
        </div>
    );
};

Card.displayName = 'Card';

Card.propTypes = {
    /** white, underlined, underlined-secondary */
    option: PropTypes.string,
    /** xs, sm, md, lg, xl */
    size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl']),
    className: PropTypes.string,
    style: PropTypes.object,
    children: PropTypes.node
};

export default Card;
