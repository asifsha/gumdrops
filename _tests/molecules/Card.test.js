/* globals mount */
import React from 'react';
import Card from '../../components/molecules/Card';

const defaultProps = {
    option: 'underlined-secondary',
    size: 'sm',
    className: 'custom',
    style: { width: 120 }
};

describe('Expect <Card>', () => {
    it('to render', () => {
        const wrapper = mount(
            <Card {...defaultProps}>
                <div>Foo</div>
            </Card>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
