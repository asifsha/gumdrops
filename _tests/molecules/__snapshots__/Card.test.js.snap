// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Card> to render 1`] = `
<Card
  className="custom"
  option="underlined-secondary"
  size="sm"
  style={
    Object {
      "width": 120,
    }
  }
>
  <div
    className="gds-card custom gds-card--underlined-secondary gds-card--sm"
    style={
      Object {
        "width": 120,
      }
    }
  >
    <div>
      Foo
    </div>
  </div>
</Card>
`;
